baseplate/frameworks/pyramid/__init__.py

```diff
@@ -64,10 +68,15 @@ def __next__(self) -> bytes:
         try:
             return next(self.app_iter)
         except StopIteration:
-            self.span.finish()
+            trace.get_current_span().set_status(trace.status.StatusCode.OK)
+            if self.span:
+                self.span.finish()
             raise
-        except:  # noqa: E722
-            self.span.finish(exc_info=sys.exc_info())
+        except Exception as e:  # noqa: E722
```

Comment:
This is a behavior change (bare except catches all exceptions, `except Exception` catches only exceptions that inherit from Exception. So, this no longer catches exceptions like `KeyboardInterrupt`. Is that OK? If you want to keep the old behavior, you can use `except BaseException as e` instead, which is equivalent to the bare except.

You should be able to drop the `noqa` here now that you're not using a bare except.

Reply:
In this case I think we are ok using `Exception`. The opentelemetry lib want's an `Exception` for it's `record_exception` function. As in all these cases we just end the span and re-raise, I think it's fine if we miss ending the span for exceptions that inherit from `BaseException` but not `Exception`.